daffodil-core/src/main/scala/org/apache/daffodil/dpath/Conversions.scala

```diff
@@ -269,74 +267,13 @@ object Conversion {
 
       case (_, Exists) => Nil
       case (_, other) => {
-        val (relevantExpr: Expression, detailMsg: String) =
-          polymorphicExpressionDiagnostics(st, tt, context)
         context.SDE(
-          "In expression %s, the type %s cannot be converted to %s.%s",
-          relevantExpr.text,
+          "The type %s cannot be converted to %s.",
```

Comment:
It's probably still worth having "In expression %s", you just need to use ``context`` to get the whole text rather than ``relevantExpr``.

Reply:
Done